### ISSUE TITLE
ONNX Run decomposition crash

### ISSUE THREAD

**Issue by Youlean** [lucidrains/rotary-embedding-torch#44]
When I do an ONNX export with torch 2.8.0, I am getting a silent crash in this step: 

[torch.onnx] Run decomposition...

-----------

I have tracked the issue, and it seems that this line in **get_seq_pos** is the problem: 

`return (torch.arange(seq_len, device = device, dtype = dtype) + offset) / self.interpolate_factor`

If you remove operations after torch.arange, everything works. 
The weird thing is that this works perfectly in Python 3.11, but crashes in Python 3.12+.

The fix is simple: you need to detach the tensor, but I am not sure if this is the correct way to do it:

`return (torch.arange(seq_len, device = device, dtype = dtype).detach() + offset) / self.interpolate_factor`

This works when training, though.
